### PR TITLE
Provision KVs for hdb and support username/password logon to hdb VM

### PR DESCRIPTION
**Pull request by YunziZhangAzure**
## Problem
<Please describe what problem this PR is trying to resolve>
#775

## Solution
<Please elaborate the solution for the problem>
1. Create two KVs for hdb.

2. Although HANA database only supports Linux VM, using password as authentication type should be supported. 
If authentication type = password, a password will be generated unless user specifies one.

3. The password will be output to the hosts.yml and hosts for Ansible use

## Tests
<Please provide steps to test the PR>

## Notes
<Additional comments for the PR>

Retrieving the ssh key from saplandscape is not included in this pr.

### CODE REVIEW

File: deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf [hunk 1]
```diff
@@ -174,9 +174,12 @@ resource "azurerm_linux_virtual_machine" "vm-dbnode" {
...
   }
 
-  admin_ssh_key {
-    username   = local.hdb_vms[count.index].authentication.username
-    public_key = file(var.sshkey.path_to_public_key)
+  dynamic "admin_ssh_key" {
+    for_each = range(local.enable_auth_password ? 0 : 1)
+    content {
+      username   = local.hdb_vms[count.index].authentication.username
+      public_key = file(var.sshkey.path_to_public_key)
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf:181]: If key is enabled, I think it should read the sshkey pair from `sap_landscape` KV.

**Review by rukawata — COMMENTED** [review]
No issue with the username/pwd logon. Just one question regarding sshkey. Or this will come in other PR?

File: deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf (hunk 1 above)
Comment (YunziZhangAzure) [reply deploy/terraform/terraform-units/modules/sap_system/hdb_node/vm-hdb.tf:181]: Ya, this will be included in a future'pr.

**Review by rukawata — DISMISSED** [review]
lgtm!

**Review by rukawata — APPROVED** [review]
thanks for fixing the bug :)

**Review by rukawata — APPROVED** [review]

**Pull request merged**